[ESLINT] Disable `any` type usage

# Expected Behavior

When using `any` as type, eslint should raise an error.

# Current Behavior

Eslint doesn't raise any error or warning.

## Reproduction

```ts
const FOO: any = "bar"
```

## Logs

```
> my-app@0.1.0 lint
> eslint "./src/**/*.ts" "./src/**/*.tsx"


/home/user/code/e-constellatio/react-template/src/App.tsx
  4:7  warning  'FOO' is assigned a value but never used  @typescript-eslint/no-unused-vars

✖ 1 problem (0 errors, 1 warning)
```

# Proposed solution

I suggest adding [`no-explicit-any`  rule](https://typescript-eslint.io/rules/no-explicit-any/) to `error` in eslint config.

# Checks

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I'm reporting the issue to the correct repository (for multi-repository projects)

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -42,6 +42,29 @@
     ],
     "@typescript-eslint/await-thenable": "error",
     "@typescript-eslint/ban-ts-comment": "error",
+    "@typescript-eslint/ban-types": [
+      "error",
+      {
+        "types": {
+          "object": {
+            "message": "In TS object mean any JS object. This type is way too broad, consider narrowing it down."
+          },
+          "Function": {
+            "message": "The Function type is too braod. Using this type will cause you trouble with readonly.",
+            "fixWith": "()=>void"
+          },
+          "CallableFunction": {
+            "message": "The CallableFunction type is too braod. Using this type will cause you trouble with readonly.",
+            "fixWith": "()=>void"
+          },
+          "NewableFunction": {
+            "message": "The NewableFunction type is too braod. Using this type will cause you trouble with readonly.",
+            "fixWith": "()=>void"
+          }
+        },
+        "extendDefaults": true
+      }
+    ],
     "@typescript-eslint/class-literal-property-style": "warn",
     "@typescript-eslint/consistent-generic-constructors": "warn",
     "@typescript-eslint/consistent-indexed-object-style": "warn",

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -243,6 +243,7 @@
     "@typescript-eslint/dot-notation": "warn",
     "@typescript-eslint/no-dupe-class-members": "warn",
     "@typescript-eslint/no-empty-function": "error",
+    "@typescript-eslint/no-explicit-any": "warn",
     "@typescript-eslint/no-implied-eval": "error",
     "@typescript-eslint/no-invalid-this": "warn",
     "@typescript-eslint/no-loop-func": "warn",

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -48,7 +48,10 @@
     "@typescript-eslint/consistent-type-assertions": "warn",
     "@typescript-eslint/consistent-type-definitions": "warn",
     "@typescript-eslint/consistent-type-exports": "warn",
-    "@typescript-eslint/consistent-type-imports": "warn",
+    "@typescript-eslint/consistent-type-imports": [
+      "warn",
+      {"fixStyle": "inline-type-imports"}
+    ],
     "@typescript-eslint/explicit-function-return-type": "warn",
     "@typescript-eslint/explicit-member-accessibility": "warn",
     "@typescript-eslint/explicit-module-boundary-types": "warn",

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -60,6 +60,10 @@
           "NewableFunction": {
             "message": "The NewableFunction type is too braod. Using this type will cause you trouble with readonly.",
             "fixWith": "()=>void"
+          },
+          "any": {
+            "message": "The any type is unsafe and too broad. Consider using a more constrained type if possible.",
+            "fixWith": "unknown"
           }
         },
         "extendDefaults": true
@@ -266,7 +270,6 @@
     "@typescript-eslint/dot-notation": "warn",
     "@typescript-eslint/no-dupe-class-members": "warn",
     "@typescript-eslint/no-empty-function": "error",
-    "@typescript-eslint/no-explicit-any": "warn",
     "@typescript-eslint/no-implied-eval": "error",
     "@typescript-eslint/no-invalid-this": "warn",
     "@typescript-eslint/no-loop-func": "warn",

--- a/.eslintrc.json
+++ b/.eslintrc.json
@@ -70,7 +70,7 @@
         "leadingUnderscore": "forbid",
         "trailingUnderscore": "forbid",
         "types": ["boolean"],
-        "prefix": ["IS_", "is_"]
+        "prefix": ["IS_", "is"]
       },
       {
         "selector": ["variable"],

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -14,7 +14,8 @@
     "resolveJsonModule": true,
     "isolatedModules": true,
     "noEmit": true,
-    "jsx": "react-jsx"
+    "jsx": "react-jsx",
+    "strictBindCallApply": true
   },
   "include": ["src"]
 }

